src/v/cloud_storage/partition_manifest.cc

```diff
@@ -312,10 +312,17 @@ partition_manifest::get_start_kafka_offset() const {
             auto delta = iter->second.delta_offset;
             return _start_offset - delta;
         } else {
-            throw std::runtime_error(fmt_with_ctx(
```

Comment:
I've been trying to rationalize this change, and I think the rule we're trying to fortify is that just because `_start_offset` moves, that doesn't mean those segment's aren't accessible. We already see this along the read path, where `segment_containing()` doesn't check that a requested offset falls below `_start_offset`. I think it makes sense to standardize that behavior.